api/models.py

```diff
@@ -117,23 +117,31 @@ def create_indexes(cls, collection):
         collection.create_index("name", unique=True)
 
 
-class User(DatabaseModel):
-    """API user model"""
+class UserProfile(BaseModel):
+    """API user profile model"""
     username: str
     hashed_password: str = Field(description="Hash of the plaintext password")
```

Comment:
I wonder if these 2 fields should rather be in the User model rather than User Profile.  I think the profile part hasn't been well defined yet.  The password will most likely require a dedicated API endpoint to update it, and maybe the username too.  In fact, even `groups` might be something to keep in `User` as it's not something the users should be able to update themselves.

Reply:
I thought we agreed to move `username`, `password`, and `groups` to user profile. Only passwords should be hidden from the response when we get all user profiles.

Reply:
Right, I guess that's not very clearly defined then.  One thing is about information users can read publicly, then what users can read from their own profile, then what can users update themselves.  Users shouldn't be able to escalate their privileges, and passwords shouldn't be readable at all (even encrypted as that facilitates brute-force attacks).